### PR TITLE
[SPARK-14952][CORE][ML] Remove methods that were deprecated in 1.6.0

### PR DESCRIPTION
**Pull request by hvanhovell**
#### What changes were proposed in this pull request?

This PR removes three methods the were deprecated in 1.6.0:
- `PortableDataStream.close()`
- `LinearRegression.weights`
- `LogisticRegression.weights`

The rationale for doing this is that the impact is small and that Spark 2.0 is a major release.
#### How was this patch tested?

Compilation succeded.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala [hunk 1]
```diff
@@ -480,9 +480,6 @@ class LogisticRegressionModel private[spark] (
   extends ProbabilisticClassificationModel[Vector, LogisticRegressionModel]
   with LogisticRegressionParams with MLWritable {
 
-  @deprecated("Use coefficients instead.", "1.6.0")
-  def weights: Vector = coefficients
-
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: Should we also remove ```weights``` at Python API?
https://github.com/apache/spark/blob/master/python/pyspark/ml/classification.py#L223

File: mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala [hunk 2]
```diff
@@ -389,9 +389,6 @@ class LinearRegressionModel private[ml] (
 
   private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
 
-  @deprecated("Use coefficients instead.", "1.6.0")
-  def weights: Vector = coefficients
-
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala]: Ditto, remove usage at Python API.

File: mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala (hunk 1 above)
Comment (hvanhovell) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: Yeah you are absolutely right about that. I'll fix that in the next pass.

**Comment by HyukjinKwon** [comment]
(I think `cc @srowen` should be removed because the PR description explains the PR itself and the names of reviewers might not be related with the PR itself. `@` will be removed by Python merge script when merging but `cc` and `srowen` will still remain, eg. https://github.com/apache/spark/commit/4514aebd1e807a665c270bfdc3f1127b3a1da898. Also I remember I was told so by one of committers)

**Comment by srowen** [comment]
@hvanhovell if you rebase I can merge this. I think it is OK.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**